tdrs-backend/tdpservice/users/api/authorization_check.py

```diff
@@ -47,3 +49,21 @@ def get(self, request, *args, **kwargs):
         else:
             logger.info("Auth check FAIL for user on %s", timezone.now())
             return Response({"authenticated": False})
+
+class KibanaAuthorizationCheck(APIView):
+    """Check if user is authorized to view Kibana."""
+
+    query_string = False
+    pattern_name = "kibana-authorization-check"
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request, *args, **kwargs):
+        """Handle get request and verify user is authorized to access kibana."""
+        user = request.user
+
+        user_in_valid_group = user.is_ofa_sys_admin or user.is_ofa_admin
```

Comment:
@elipe17 given this:

> All users who can access Kibana will be able to read/write any index in Elastic. 

we want to lock this down to sys admin role, as this seems like privileged access. 

I'm also now wondering what [this permission group](https://github.com/raft-tech/TANF-app/issues/1441) will have access to in the way of parsed data:
- Might there be a feasible way for sys admins to pass parsed data to members of this less privileged group? 
- I also thought we might have some queries pre-built in Kibana that would enable users with access to it to quickly generate extracts (e.g. .csv or .json format)  of parsed data (e.g. all T1 records for FY2024, Q1), which might help mitigate risks of changing anything in the db. 

I'm mostly thinking aloud here...maybe we have to get creative given the limitations of working with this wrapper.

Reply:
Updating so that OFA System Admin are the only users who can access Kibana. Until we get more control over our cloud resources Kibana's access is all or nothing. Users who can access admin will have admin privileges in Kibana. However, if the users delete data/documents/indices in Kibana, that will not affect Postgres and no data will be truly lost. We can recover the data in Elastic/Kibana by rebuilding the search indices with elastic_search_dsl manage.py command.

Tickets will be written to have pre-canned dashboards and queries so that Kibana is just ready when deployed versus having to do manual configuration.